### PR TITLE
feat: adding opsgenie on call schedule fetching functions

### PR DESCRIPTION
**Pull request by Charles546**

### CODE REVIEW

**Comment by Charles546** [comment]
should definitely squash this

File: opsgenie.yaml [hunk 1]
```diff
@@ -84,6 +84,142 @@ systems:
...
+
+          notes:
+            - See below for example
+            - example: |
+                ---
+                workflows:
+                  steps:
+                    - call_function: opsgenie.schedules
+
+      who-is-oncall:
```
Comment (wyardley) [inline opsgenie.yaml]: maybe remove the dashes to keep it consistent with their command?

File: opsgenie.yaml [hunk 2]
```diff
@@ -84,6 +84,142 @@ systems:
                       call_function: opsgenie.snooze
                       #  alert_Id is exported from the event
 
+      schedules:
+        driver: web
+        rawAction: request
+        parameters:
+          URL: https://api.opsgenie.com/v2/schedules{{ with .ctx.scheduleId }}/{{ . }}?identifierType={{ default "name" $.ctx.scheduleIdType }}{{ end }}
```
Comment (wyardley) [inline opsgenie.yaml]: Are there retries on these? What happens in the case where a single API call fails?

File: opsgenie.yaml (hunk 1 above)
Comment (Charles546) [reply opsgenie.yaml]: done

**Pull request merged**